src/issuer/SsiCredentialIssuer.Service/BusinessLogic/IssuerBusinessLogic.cs

```diff
@@ -0,0 +1,510 @@
+    }
+
+    public async Task<Guid> CreateFrameworkCredential(CreateFrameworkCredentialRequest requestData, CancellationToken cancellationToken)
+    {
+        var companyCredentialDetailsRepository = _repositories.GetInstance<ICompanySsiDetailsRepository>();
+        var result = await companyCredentialDetailsRepository.CheckCredentialTypeIdExistsForExternalTypeDetailVersionId(requestData.UseCaseFrameworkVersionId, requestData.UseCaseFrameworkId).ConfigureAwait(false);
+        if (!result.Exists)
+        {
+            throw ControllerArgumentException.Create(CredentialErrors.EXTERNAL_TYPE_DETAIL_NOT_FOUND, new ErrorParameter[] { new("verifiedCredentialExternalTypeDetailId", requestData.UseCaseFrameworkId.ToString()) });
+        }
+
+        if (result.Expiry < _dateTimeProvider.OffsetNow)
+        {
+            throw ControllerArgumentException.Create(CredentialErrors.EXPIRY_DATE_IN_PAST);
+        }
+
+        if (string.IsNullOrWhiteSpace(result.Version))
+        {
+            throw ControllerArgumentException.Create(CredentialErrors.EMPTY_VERSION);
+        }
+
+        if (string.IsNullOrWhiteSpace(result.Template))
+        {
+            throw ControllerArgumentException.Create(CredentialErrors.EMPTY_TEMPLATE);
+        }
+
+        if (result.UseCase.Count() != 1)
+        {
+            throw ControllerArgumentException.Create(CredentialErrors.MULTIPLE_USE_CASES);
+        }
+
+        var useCase = result.UseCase.Single();
+        var holderDid = await GetHolderInformation(requestData.Holder, cancellationToken).ConfigureAwait(false);
+        var schemaData = new FrameworkCredential(
+            Guid.NewGuid(),
+            Context,
+            new[] { "VerifiableCredential", $"{useCase}Credential" },
+            $"{useCase}Credential",
+            $"Framework Credential for UseCase {useCase}",
+            DateTimeOffset.UtcNow,
+            result.Expiry,
+            _settings.IssuerDid,
+            new FrameworkCredentialSubject(
+                holderDid,
+                requestData.HolderBpn,
+                "UseCaseFramework",
+                useCase,
+                result.Template!,
+                result.Version!
+            ),
+            new CredentialStatus(
+                _settings.StatusListUrl,
+                StatusList)
+        );
+        var schema = JsonSerializer.Serialize(schemaData, Options);
+        return await HandleCredentialProcessCreation(_identity.Bpnl, VerifiedCredentialTypeKindId.FRAMEWORK, requestData.UseCaseFrameworkId, schema, requestData.TechnicalUserDetails, requestData.UseCaseFrameworkVersionId, requestData.CallbackUrl, companyCredentialDetailsRepository);
+    }
+
+    private async Task<string> GetHolderInformation(string didDocumentLocation, CancellationToken cancellationToken)
+    {
+        var client = _clientFactory.CreateClient("didDocumentDownload");
+        var result = await client.GetAsync(HttpUtility.HtmlEncode(didDocumentLocation), cancellationToken)
```

Comment:
HtmlEncode does translate special characters to their html-representation. E.g. &amp; instead of &. This is not suitable for Urls

Reply:
adjusted the validation of the url

Reply:
adjusted the validation of the url